azblob/reader.go

```diff
@@ -35,38 +45,42 @@ type Reader interface {
 //	container: merklebuilder
 func NewReaderNoAuth(accountName string, url string, container string) (Reader, error) {
 	logger.Sugar.Infof(
-		"New Reader with accountName: %s, for container: %s",
-		accountName, container,
+		"New Reader for url: %s, with accountName: %s, for container: %s",
+		url, accountName, container,
 	)
 
 	var err error
-
-	if accountName == "" || url == "" {
-		return nil, errors.New("missing connection configuration variables")
+	if url == "" {
+		return nil, errors.New("url is a required parameter and cannot be empty")
 	}
 
 	azp := &Storer{
-		AccountName:   accountName,
-		ResourceGroup: "", // just for logging
-		Subscription:  "", // just for logging
+		AccountName:   accountName, // just for logging
+		ResourceGroup: "",          // just for logging
+		Subscription:  "",          // just for logging
 		Container:     container,
 		credential:    nil,
 		rootURL:       url,
 	}
-
-	azp.containerURL = fmt.Sprintf(
-		"%s%s",
-		url,
-		container,
-	)
 	azp.serviceClient, err = azStorageBlob.NewServiceClientWithNoCredential(
 		url,
 		nil,
 	)
 	if err != nil {
-		logger.Sugar.Infof("unable to create serviceclient %s: %v", azp.containerURL, err)
+		logger.Sugar.Infof("unable to create serviceclient %s: %v", url, err)
 		return nil, err
 	}
+
+	if container == "" {
+		logger.Sugar.Infof("container not provided, container client not created")
+		return nil, nil
```

Comment:
shouldn't that return an error?

Reply:
The usage here is when you expect to just use the storage client. That is the FilterBlobs method right now.